dataset/src/main/scala/frameless/TypedDataset.scala

```diff
@@ -659,6 +668,43 @@ class TypedDataset[T] protected[frameless](val dataset: Dataset[T])(implicit val
     * @see [[frameless.TypedDataset#project]]
     */
   def drop[U](implicit projector: SmartProject[T,U]): TypedDataset[U] = project[U]
+  
+  /**
+    * Returns a `QueryExecution` from this [[TypedDataset]].
+    *
+    * It is the primary workflow for executing relational queries using Spark.  Designed to allow easy
+    * access to the intermediate phases of query execution for developers.
+    *
+    * Differs from `Dataset#queryExecution` by wrapping it's result into an effect-suspending `F[_]`.
+    *
+    * apache/spark
+    */
+  def queryExecution[F[_]](implicit F: SparkDelay[F]) : F[QueryExecution] =
+    F.delay(dataset.queryExecution)
+
+  /**
+    * Returns the schema of this [[TypedDataset]].
+    *
+    * Differs from `Dataset#schema` by wrapping it's result into an effect-suspending `F[_]`.
+    */
+  def schema[F[_]](implicit F: SparkDelay[F]): F[StructType] =
```

Comment:
I believe this might exist already https://github.com/typelevel/frameless/blob/820d1f737dd452d37b11f4d2b69a0a9df3f21f82/dataset/src/main/scala/frameless/TypedDatasetForwarded.scala#L28